polkadot/runtime/parachains/src/scheduler.rs

```diff
@@ -395,15 +398,17 @@ impl<T: Config> Pallet<T> {
 					// the back of the claimqueue.
 					for drop in dropped_claims {
 						match T::AssignmentProvider::pop_assignment_for_core(core_idx, drop) {
-							Some(assignment) => {
+							Some(assignment)
```

Comment:
Leaving this as a defensive check might make sense, but I feel like this should be checked earlier in the process. E.g. for on-demand an order should be deemed invalid if rent has not been paid.